DS13/code/modules/overmap/components.dm

```diff
@@ -0,0 +1,643 @@
+	desc = "This console gives you the power to control a starship."
+	icon_state = "pilot"
+	position = "pilot"
+	req_access = list(ACCESS_HEADS)
+
+/obj/structure/overmap_component/science
+	name = "Science station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "science"
+	position = "science"
+	req_access = list(ACCESS_RESEARCH)
+
+/obj/structure/overmap_component/tactical
+	name = "Weapons station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "tactical"
+	position = "tactical"
+	req_access = list(ACCESS_SECURITY)
+
+/obj/structure/overmap_component/helm/miranda
+	name = "Piloting station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "pilot"
+	position = "pilot"
+	id = "miranda"
+
+/obj/structure/overmap_component/science/miranda
+	name = "Science station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "science"
+	position = "science"
+	id = "miranda"
+
+/obj/structure/overmap_component/tactical/miranda
+	name = "Weapons station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "tactical"
+	position = "tactical"
+	id = "miranda"
+
+/obj/structure/overmap_component/helm/warbird
+	name = "Piloting station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "pilot"
+	position = "pilot"
+	id = "warbird"
+
+/obj/structure/overmap_component/science/warbird
+	name = "Science station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "science"
+	position = "science"
+	id = "warbird"
+
+/obj/structure/overmap_component/tactical/warbird
+	name = "Weapons station"
+	desc = "This console gives you the power to control a starship."
+	icon_state = "tactical"
+	position = "tactical"
+	id = "warbird"
+
+///INJECTORS, RELAYS, AND INTEGRITY FIELD GEN!///
+// Each relay gives +2 max power per system
+
+
+/obj/structure/overmap_component/plasma_injector
+	name = "Plasma injector"
+	desc = "This sturdy device will inject supplied plasma into the ship's ODN relays, allowing for massive power transference. Simply load it with a plasma canister, or click it with an empty hand to remove a canister"
+	icon_state = "injector"
+	req_access = list(ACCESS_ENGINE_EQUIP)
+	var/obj/item/tank/internals/plasma/loaded_tank = null //Nicked from rad collector code, see there for documentation (collector.dm)
+	max_integrity = 350
+	integrity_failure = 50
+	var/powerproduction_drain = 0.015 //Relatively high drain on the plasma tanks.
+	var/drainratio = 1
+	var/supplying = FALSE //Are we supplying plasma to another object?
+	var/obj/structure/overmap_component/plasma_relay/supply_to
+	var/obj/structure/overmap_component/integrity_field_generator/generator
+	var/locked = FALSE
+
+/obj/structure/overmap_component/plasma_injector/attack_hand(mob/user)
+	var/mob/living/carbon/human/L = user
+	var/obj/item/card/id/ID = L.wear_id
+	if(ID && istype(ID))
```

Comment:
You already have locking & unlocking this via hitting it via your ID, can you not use that system instead of having an extra check for ID?